src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PrefixMatchPackageSourceNames.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.Configuration
+{
+    public class PrefixMatchPackageSourceNames
+    {
+        /// <summary>
+        /// Generates a <see cref="PrefixMatchPackageSourceNames"/> based on the constructor params.
+        /// </summary>
+        /// <param name="packageNamespaceSectionPresent">If package namespace section present in nuget.config file</param>
+        /// <param name="packageSourceNames">Package source names with prefixes matching search term</param>
+        public PrefixMatchPackageSourceNames(bool packageNamespaceSectionPresent, HashSet<string> packageSourceNames)
+        {
+            PackageNamespaceSectionPresent = packageNamespaceSectionPresent;
+            PackageSourceNames = packageSourceNames;
+        }
+
+        public bool PackageNamespaceSectionPresent { get; }
+        public HashSet<string> PackageSourceNames { get; }
```

Comment:
I think we should return a collection of `package source URLs` instead of `source names` in the result. If users specify `-Source` parameter, then `URLs are passed` to the command.

Reply:
Then how the mapping between `packageSource key` and `value` works?
Unless we already have exact values `nuget.com ` and `-Source` then it's not going to find each other.
If it already have that exact values then passing some urls with `-Source` doesn't change anything for input arguments.